src/test/java/com/github/dariobalinzo/task/ElasticSourceTaskTest.java

```diff
@@ -150,9 +150,61 @@ public void shouldRunSourceTaskWithJsonCastFilter() throws IOException, Interrup
         //when (fetching first page)
         task.start(conf);
         List<SourceRecord> poll1 = task.poll();
-        assertEquals("Struct{fullName=\"Test\",age=10,ts=111}", poll1.get(0).value().toString());
+        assertEquals("Struct{fullName=\"Test\",nonavrofield=non-avro-field,avroField=avro-field,age=10,ts=111}", poll1.get(0).value().toString());
 
         task.stop();
     }
 
+    @Test
+    public void shouldRunSourceTaskWithAvroNameConverter() throws IOException, InterruptedException {
+        //given
+        deleteTestIndex();
+
+        insertMockData(111);
+        insertMockData(112);
+        insertMockData(113);
+        insertMockData(114);
+        refreshIndex();
+
+        ElasticSourceTask task = new ElasticSourceTask(ElasticSourceConnectorConfig.AVRO_FIELD_CONVERTER);
```

Comment:
Avro config should be standard in the kafka. Since the connector works also in cluster/multi-node, the class ElasticSourceTask will be instantiated by kafka framework using reflection. We cannot control the class creation. All the config should be passed in the method start (like line 176 of this file). Since the config is a map, you can override it in this single test (like the test for whitelist in this file).